pkg/build/controller/build/build_controller.go

```diff
@@ -2023,39 +2040,42 @@ func (bc *BuildController) getAdditionalTrustedCAData(config *configv1.Image) (m
 	return additionalCA.Data, nil
 }
 
-func (bc *BuildController) createBuildRegistriesConfigData(config *configv1.Image) (string, error) {
-	registriesConfig := config.Spec.RegistrySources
-	if len(registriesConfig.InsecureRegistries) == 0 && len(registriesConfig.BlockedRegistries) == 0 {
+func (bc *BuildController) createBuildRegistriesConfigData(config *configv1.Image, policies []*operatorv1alpha1.ImageContentSourcePolicy) (string, error) {
+
+	blockedRegs := []string{}
+	insecureRegs := []string{}
+	if config != nil {
+		insecureRegs = config.Spec.RegistrySources.InsecureRegistries
+		blockedRegs = config.Spec.RegistrySources.BlockedRegistries
+	}
+	if len(insecureRegs) == 0 && len(blockedRegs) == 0 {
```

Comment:
(I know I’ve pointed at this line before…) This will cause ICSP to be ignored (= disconnected not to work) if this branch is taken.

Reply:
yes you did .... some stuff got lost in the various rebases ... will address in next push

Reply:
I do know I had a version as some point that had ICSP factored in here

Reply:
ok pushed the update for this ... now `if len(insecureRegs) == 0 && len(blockedRegs) == 0 && len(policies) == 0 {`